Doc/library/functools.rst

```diff
@@ -358,6 +358,41 @@ The :mod:`functools` module defines the following functions:
       >>> basetwo('10010')
       18
 
+   If :data:`Placeholder` sentinels are present in *args*, they will be filled first
+   when :func:`partial` is called. This allows custom selection of positional arguments
+   to be pre-filled when constructing a :ref:`partial object <partial-objects>`.
+
+   If :data:`!Placeholder` sentinels are used, all of them must be filled at call time:
+
+      >>> from functools import partial, Placeholder
+      >>> say_to_world = partial(print, Placeholder, Placeholder, "world!")
+      >>> say_to_world('Hello', 'dear')
+      Hello dear world!
+
+   Calling ``say_to_world('Hello')`` would raise a :exc:`TypeError`, because
+   only one positional argument is provided, while there are two placeholders
+   in :ref:`partial object <partial-objects>`.
+
+   When successively using :func:`partial` existing :data:`!Placeholder`
+   sentinels are filled first. A place for positional argument is retained
+   when :data:`!Placeholder` sentinel is replaced with a new one:
```

Comment:
It simply means that if there was a sentinel in input partial object, one can propagate it by overriding it with the new one.

I made some changes in that whole area. Hopefully it will be better.